### ISSUE TITLE
Is onnxruntime_providers_shared needed for CPU build?

### ISSUE THREAD

**Issue by horenmar** [microsoft/onnxruntime#19487]
### Describe the issue

I noticed that prebuilt CPU packages for Windows (and only for Windows) contain `onnxruntime_providers_shared.*` alongside `onnxruntime.*`. Based on the [docs here](https://onnxruntime.ai/docs/build/eps.html#execution-provider-shared-libraries) and the fact that they are not also packaged with Linux and OSX builds, I suspect that they are not needed.

### To reproduce

Download release packages from https://github.com/microsoft/onnxruntime/releases/tag/v1.17.0 and compare the contents of e.g. onnxruntime-win-x64-1.17.0.zip with onnxruntime-linux-x64-1.17.0.tgz.

### Urgency

Not urgent

### Platform

Windows

### OS Version

N/A

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.17.0

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

Default CPU

**Comment by snnn** [comment]
onnxruntime_providers_shared.dll is only used by some execution providers(e.g. CUDA). The default CPU execution providers doesn't need it.

**Comment by emaschino** [comment]
I'm confused on this too. The documentation linked by @horenmar states that the shared providers depend on onnxruntime_providers_shared. I built ONNX Runtime with DirectML (on Windows UWP) and with NNAPI (on Android) EPs as shared libraries. None of the generated binaries have a dependency on onnxruntime_providers_shared. This can be checked on Windows with dumpbin /DEPENDENTS for example. And I've checked that my client app is working fine with these binaries, without providing onnxruntime_providers_shared.
